components/SearchBar/index.js

```diff
@@ -0,0 +1,201 @@
+import { useState } from "react";
+import TransactionCard from "../TransactionCard/index";
+import {
+  StyledFilterButton,
+  StyledFilterContainer,
+  StyledFilterItem,
+  StyledSearchAndFilter,
+  StyledSearchContainer,
+  StyledSearchInput,
+  StyledSearchInputSelect,
+  StyledSearchUl,
+} from "./SearchBar.styled";
+
+function SearchBar({ transactions, search, onSearch, isSearchEntry }) {
```

Comment:
The name of this component is misleading, since it not only diesplays the searchbar but also the filters and the search results. I would advise you to change the structure a bit: display the search results inside the index.js page and remove them from this component. You can filter your transactions directly in `pages/index.js` with your nice filter function. Then this components job is to only get search and filter information and set them to the global state that you already set up. The display of the search results is simply your normal transactions list already present ion the homepage with a filtered array as an input.

Reply:
we have now changed that and rewritten it as you suggested.